### DISCUSSION TITLE
Introduction to LibKTX with real example games

### DISCUSSION THREAD

**Discussion by Quillraven** [libktx/ktx#328] [Show and tell]
Hey guys,

I am Simon, one of the contributors to LibKTX. I have made a few projects already with LibKTX and also tutorials that hopefully help other developers out there to getting started with games using LibKTX.

Here are the things that might help you:
- [A Simple KTX Game](https://github.com/Quillraven/SimpleKtxGame): this is a tutorial on how to convert [A simple game](https://github.com/libgdx/libgdx/wiki/A-simple-game) from the official LibGDX wiki to Kotlin using LibKTX. Checkout the [wiki](https://github.com/Quillraven/SimpleKtxGame/wiki) for the details.
One thing to note: I would recommend to use [gdx-liftoff](https://github.com/tommyettinger/gdx-liftoff) as a setup. In my opinion it produces better gradle build files than the official LibGDX setup.
- [LibGDX Kotlin tutorial using LibKTX](https://www.youtube.com/watch?v=7aa6CY7_j9U&list=PLTKHCDn5RKK-8lZmjZoG4rFywN_SLbZR8&index=1&t=6s&ab_channel=Quillraven): if you already have some experience with LibGDX then you might be interested in this video tutorial series. You will develop a game where the player needs to select several powerups to escape the dangerous dark matter with his spaceship. The game includes everything that you will need for 2D-game development including things like Shaders, Sounds, Graphics, Input, Event Handling and many other useful things.
- [Quilly's Adventure](https://github.com/Quillraven/Quilly-s-Adventure): this is an example project for an adventure like RPG using LibKTX and some selfmade Kotlin extensions. Besides the things from the two projects above, you can lookup how to use Box2D with LibKTX and how to define your own Kotlin DSLs for your games like monster type definitions or event triggers with conditions and actions. This is only the source code but the [wiki](https://github.com/Quillraven/Quilly-s-Adventure/wiki) contains some information on how to use the framework which might be helpful. Also, if you went through the video tutorial series mentioned above then you will also understand this codebase because it is very similar.
- [Quilly Crawler](https://github.com/Quillraven/Quilly-Crawler): this is another example project for a dungeon crawler game using LibKTX, Box2D and Ashley. It contains a **commons** module which contains some generic solutions for any game.

I hope that these things help some people out there :) 

Happy coding!

**Comment by czyzby** [comment]
Very useful resources, @Quillraven! I think your games are among the best open source KTX projects out there. Thanks for sharing.

**Reply by czyzby** [reply]
@Quillraven Would you mind adding me as a collaborator to the [SimpleKtxGame](https://github.com/Quillraven/SimpleKtxGame)? I might just do another section on coroutines-based `AssetStorage` some time. By the way, you've received some praise for the tutorial on libGDX discord. :)

**Reply by Quillraven** [reply]
You should be a collaborator now for SimpleKtxGame! Thank you for letting me know about the praise. I also get some comments from time to time under my youtube videos. Glad that it seems to be useful and people are liking it :)

**Reply by czyzby** [reply]
And here's [another KTX game by Quillraven: Dinoleon](https://github.com/Quillraven/Dinoleon), using the [Fleks](https://github.com/Quillraven/Fleks) Entity-Component-System library. Make sure to check it out! :)

**Comment by steelx** [comment]
thank you :) Simoo00n

**Comment by andrzej-nov** [comment]
Here is one more Open Source KTX game, [Tangler](https://github.com/andrzej-nov/Tangler). A 2D graphics puzzle with dynamically generated sprite textures. The code is thoroughly commented.

**Comment by jyjeanne** [comment]
Thanks for your work, Simoooon!
